vlasiator.cpp

```diff
@@ -924,25 +929,36 @@ int main(int argn,char* args[]) {
          else {
             doNow[0] = 0;
          }
-         if (globalflags::balanceLoad == true) {
+         if (globalflags::balanceLoad || globalflags::doRefine) {
             doNow[1] = 1;
             globalflags::balanceLoad = false;
+            if (globalflags::doRefine) {
+               doNow[2] = 1;
+               globalflags::doRefine = false;
+            }
          }
       }
-      MPI_Bcast( &doNow, 2 , MPI_INT , MASTER_RANK ,MPI_COMM_WORLD);
+      MPI_Bcast( &doNow, 3 , MPI_INT , MASTER_RANK ,MPI_COMM_WORLD);
       writeRestartNow = doNow[0];
       doNow[0] = 0;
       if (doNow[1] == 1) {
          P::prepareForRebalance = true;
          doNow[1] = 0;
       }
+      if (doNow[2]) {
+         refineNow = true;
+         doNow[2] = false;
```

Comment:
I think we should also set overrideRebalanceNow=true here so that the refinement happens actually right now instead of at the next LB.

Reply:
doNow[2] also sets doNow[1] so it prepares for rebalance and goes through that first. iirc that requires one timestep for calculating weights.